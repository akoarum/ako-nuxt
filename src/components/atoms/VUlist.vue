<template>
  <ul :class="styledClass" class="list">
    <template v-for="item in items">
      <li :key="item.id" class="list__item">{{ item.label }}</li>
    </template>
  </ul>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { ListItem } from '~/types/models'

export default Vue.extend({
  props: {
    items: { type: Array as PropType<ListItem[]>, required: true },
    ordered: { type: Boolean }
  },
  computed: {
    styledClass(): string {
      if (this.ordered) return '-ordered'
      return ''
    }
  }
})
</script>

<style lang="scss" scoped>
@import './styles/list';
</style>
