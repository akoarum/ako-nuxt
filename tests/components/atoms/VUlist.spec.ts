import { shallowMount } from '@vue/test-utils'
import VUlist from '~/components/atoms/VUlist.vue'

describe('VUlist', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VUlist, {
      propsData: {
        items: [
          { id: 1, label: 'テスト' },
          { id: 2, label: 'テスト2' }
        ]
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClass: propsからクラスを生成して返す', () => {
      expect(wrapper.vm.styledClass).toBe('')
      wrapper.setProps({ ordered: true })
      expect(wrapper.vm.styledClass).toBe('-ordered')
    })
  })
})
