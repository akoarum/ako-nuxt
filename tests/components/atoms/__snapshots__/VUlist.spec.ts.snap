// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUlist initialized snapshot 1`] = `
<ul
  class="list"
>
  <li
    class="list__item"
  >
    テスト
  </li>
  <li
    class="list__item"
  >
    テスト2
  </li>
</ul>
`;
